contracts/SimpleOperations.sol

```diff
@@ -45,121 +46,258 @@ contract SimpleOperations is Operations {
+		public
+		view
+		returns (uint8)
+	{
+		if (clientExists(_client)) {
+			return client[_client].release[_release].track;
+		} else {
+			return 0;
+		}
 	}
 
-	function latestInTrack(bytes32 _client, uint8 _track) public view returns (bytes32) {
-		return client[_client].current[_track];
+	function latestInTrack(bytes32 _client, uint8 _track)
+		public
+		view
+		returns (bytes32)
+	{
+		if (clientExists(_client)) {
+			return client[_client].current[_track];
+		} else {
+			return 0;
+		}
 	}
 
-	function build(bytes32 _client, bytes32 _checksum) public view returns (bytes32 o_release, bytes32 o_platform) {
-		var b = client[_client].build[_checksum];
-		o_release = b.release;
-		o_platform = b.platform;
+	function build(bytes32 _client, bytes32 _checksum)
+		public
+		view
+		returns (bytes32 o_release, bytes32 o_platform)
+	{
+		if (clientExists(_client)) {
+			Build storage b = client[_client].build[_checksum];
+			o_release = b.release;
+			o_platform = b.platform;
+		} else {
+			o_release = bytes32(0);
+			o_platform = bytes32(0);
+		}
 	}
 
-	function release(bytes32 _client, bytes32 _release) public view returns (uint32 o_forkBlock, uint8 o_track, uint24 o_semver, bool o_critical) {
-		var b = client[_client].release[_release];
-		o_forkBlock = b.forkBlock;
-		o_track = b.track;
-		o_semver = b.semver;
-		o_critical = b.critical;
+	function release(bytes32 _client, bytes32 _release)
+		public
+		view
+		returns (
+			uint32 o_forkBlock,
+			uint8 o_track,
+			uint24 o_semver,
+			bool o_critical
+		)
+	{
+		if (clientExists(_client)) {
+			Release storage r = client[_client].release[_release];
+			o_forkBlock = r.forkBlock;
+			o_track = r.track;
+			o_semver = r.semver;
+			o_critical = r.critical;
+		} else {
+			o_forkBlock = 0;
+			o_track = 0;
+			o_semver = 0;
+			o_critical = false;
+		}
 	}
 
-	function checksum(bytes32 _client, bytes32 _release, bytes32 _platform) public view returns (bytes32) {
-		return client[_client].release[_release].checksum[_platform];
+	function checksum(bytes32 _client, bytes32 _release, bytes32 _platform)
+		public
+		view
+		returns (bytes32)
+	{
+		if (clientExists(_client)) {
+			return client[_client].release[_release].checksum[_platform];
+		} else {
+			return bytes32(0);
+		}
 	}
 
-	function latestFork() public view returns (uint32) {
+	function latestFork()
+		public
+		view
+		returns (uint32)
+	{
 		return latestFork;
 	}
 
-	function clientOwner(address _owner) public view returns (bytes32) {
+	function clientOwner(address _owner)
+		public
+		view
+		returns (bytes32)
+	{
 		return clientOwner[_owner];
 	}
 
+	// Internals
+
+	function clientExists(bytes32 _client)
+		internal
+		view
+		returns (bool)
+	{
+		return client[_client].owner != 0;
+	}
+
+	function releaseExists(bytes32 _client, bytes32 _release)
+		internal
+		view
+		returns (bool)
+	{
+		Release storage r = client[_client].release[_release];
+		return r.forkBlock != 0 && r.track != 0 && r.semver != 0;
```

Comment:
Seems that we require forkBlock to be always set to a non-zero value. I don't think it's intuitive, better to ditch that check imho.